goflags.go

```diff
@@ -351,26 +382,71 @@ func (flagSet *FlagSet) usageFunc() {
 
 	writer := tabwriter.NewWriter(cliOutput, 0, 0, 1, ' ', 0)
 
-	flagSet.flagKeys.forEach(func(key string, data *flagData) {
-		currentFlag := flag.CommandLine.Lookup(key)
+	if len(flagSet.groups) > 0 {
+		var otherOptions []string
 
-		dataHash := data.Hash()
-		if _, ok := hashes[dataHash]; ok {
-			return // Don't print the value if printed previously
+		for _, group := range flagSet.groups {
+			fmt.Fprintf(cliOutput, "%s:\n", strings.ToUpper(group.description))
+
+			flagSet.flagKeys.forEach(func(key string, data *FlagData) {
```

Comment:
Please, PLEASE do not write such long methods. Splitting it to smaller chunks with a well choosen method name would increase the readability significantly.